### PR TITLE
Site: Add commit message guidelines for contributors (Stamatis Zampetakis)

### PR DESCRIPTION
**Pull request by zabetak**
Add guidelines for commit messages proposed by Julian Hyde in various discussions in the dev list.

### CODE REVIEW

File: site/develop/index.md [hunk 1]
```diff
@@ -83,6 +83,26 @@ If your change had multiple commits, use `git rebase -i master` to
...
+("Added a handler ...") or present tense ("Adds a handler ...").
+ * If possible, describe the user-visible behavior that you changed
+("FooCommand now creates directory if it does not exist"), rather than
+the implementation ("Add handler for FileNotFound").
+ * If you are fixing a bug, it is sufficient to describe the bug
+ ("NullPointerException if user is unknown") and people will correctly
+ surmise that the purpose of your change is to fix the bug.
+ * If you are not a committer, add your name in parentheses at the end
+ of the message.
+ * Read the messages of previous commits, and follow their style.
```
[the remark this answers is not in the document]
Comment (zabetak) [reply site/develop/index.md]: I agree. I made the change!

**Comment by asereda-gs** [comment]
@zabetak can you pls squash 2 commits and I will merge this PR.

**Comment by zabetak** [comment]
@asereda-gs done!

**Comment by zabetak** [comment]
Thanks @asereda-gs  :)

**Pull request merged**